backend/msgpack/writer.go

```diff
@@ -91,6 +91,8 @@ func (w *writer) Write(
 	mu unaggregated.MetricUnion,
 	pl policy.PoliciesList,
 ) error {
+	w.metrics.idLengths.RecordValue(float64(len(mu.ID)))
```

Comment:
This is called on the critical path of writing every metric, are we worried about the CPU overhead of doing a binary search and recording it?

Reply:
I couldn't find any tally benchmarks for recording a histogram so I made a quick one myself, recording a value took on average 35ns. So I think a fair estimate for the time these two function calls take in total would be 100ns. Even if a collector is processing 100,000 metrics per second, that's only an additional 10ms of CPU time. With that being said, just by the law of the large numbers, even if we only record metrics for a fraction of the metrics we see, we should still get a very close approximation to the true distribution of id lengths and number of policies, so I'll add a dice here so we can control what fraction of metrics we record these values for. I think these would be value generally so I'm hesitant to not record them at all.